crates/net/network/src/network.rs

```diff
@@ -48,6 +51,7 @@ impl NetworkHandle {
             local_peer_id,
             peers,
             network_mode,
+            is_syncing: Arc::new(Default::default()),
```

Comment:
what's the reason to replicate the state as opposed to forwarding the request to the provider?

Reply:
> forwarding the request to the provider?

I don't understand this, which `provider` to you mean here?

Reply:
`SyncStateProvider
`

Reply:
> as opposed to forwarding the request to the provider?

`NetworkHandle {is_syncing}` is the `SyncStateProvider`, so I'm not following here.

we track the state as AtomicBool so we can easily access this from the `NetworkManager` and eventually the sessions as well.

Reply:
ah, i thought `NetworkHandle` is querying the `SyncStateProvider`. k, i see